torch_dicom/preprocessing/crop.py

```diff
@@ -383,13 +383,30 @@ def get_bounds(self, x: Tensor, sopuid: SOPUID) -> Tensor:
             raise ValueError(f"Expected input to have at exactly 3 dimensions, got {x.ndim}")
 
         H, W = x.shape[-2:]
+        H_min, W_min = self.min_size
 
         if sopuid not in self.df.index:
-            # Select a random crop of size `self.min_size` anywhere in the image
-            h_start = torch.randint(0, H - self.min_size[0] + 1, (1,))
-            w_start = torch.randint(0, W - self.min_size[1] + 1, (1,))
-            h_end = h_start + self.min_size[0]
-            w_end = w_start + self.min_size[1]
+            non_zero_indices = torch.nonzero(x)
+            if non_zero_indices.numel():
+                # Try to find a crop that contains a nonzero pixel
+                center = non_zero_indices[torch.randint(0, len(non_zero_indices), (1,))]
```

Comment:
I think this is a pretty good approach. Perhaps we could consider other strategies in the future. E.g., out of three random crops, choose the one with highest entropy.